### PR TITLE
Add REST implementation for `MultipleValidatorStatus`

### PR DESCRIPTION
**Pull request by nalepae**
### What type of PR is this?
Feature

### Related gitHub issue:
https://github.com/prysmaticlabs/prysm/issues/11580

### What does this PR do? Why is it needed?
This adds an implementation for the `MultipleValidatorStatus` validator client method by calling the Beacon Node REST API instead of the gRPC one.

### Notes for reviewers:
I tried to mimic https://github.com/prysmaticlabs/prysm/blob/develop/beacon-chain/rpc/prysm/v1alpha1/validator/status.go#L55-L98.

A list of validators keys and validator indexes are given in inputs.
- If the input validator key corresponds to an existing validator, then nothing fancy is done.
- If the input validator key does not correspond to an existing validator, then the corresponding validator status is set as `unknown` and the corresponding validator index is set at the highest possible `uint64` value.
- If the input validator index does not correspond to an existing validator, then the corresponding validator is dropped.

Compared to gRPC version, I did not keep the same order in the outputs.
I checked in the codebase where `MultipleValidatorStatus` is used, and did not noticed any need for a specific output order.

### CODE REVIEW

File: validator/client/beacon-api/status.go [hunk 1]
```diff
@@ -11,69 +11,158 @@ import (
...
+	totalLen := len(inPubKeys) + len(inIndexes)
 
-	// Set Status
-	status, ok := beaconAPITogRPCValidatorStatus[validatorContainer.Status]
-	if !ok {
-		return nil, errors.New("invalid validator status: " + validatorContainer.Status)
-	}
+	outPubKeys := make([][]byte, totalLen)
+	outIndexes := make([]types.ValidatorIndex, totalLen)
+	validatorsStatus := make([]*ethpb.ValidatorStatusResponse, totalLen)
```
Comment (rkapka) [inline validator/client/beacon-api/status.go:51]: Let's move these variables right before the `for` loop because that's where you start using these them.

**Review by rkapka — CHANGES_REQUESTED** [review]

File: validator/client/beacon-api/status.go [hunk 2]
```diff
@@ -11,69 +11,158 @@ import (
...
 		if err != nil {
-			return nil, errors.Wrap(err, "failed to get state validators")
+			return nil, nil, nil, errors.Wrap(err, "failed to parse activation epoch")
 		}
 
-		data := activeStateValidators.Data
+		validatorStatus.ActivationEpoch = types.Epoch(activationEpoch)
 
-		var lastActivatedValidatorIndex uint64 = 0
+		// Set PositionInActivationQueue
```
Comment (rkapka) [inline validator/client/beacon-api/status.go:118]: Let's extract this to a private function `setPositionInActivationQueue`

File: validator/client/beacon-api/status.go [hunk 3]
```diff
@@ -11,69 +11,158 @@ import (
...
+		switch status {
+		case ethpb.ValidatorStatus_DEPOSITED, ethpb.ValidatorStatus_PENDING, ethpb.ValidatorStatus_PARTIALLY_DEPOSITED:
+			if !isLastActivatedValidatorIndexRetrieved {
+				isLastActivatedValidatorIndexRetrieved = true
```
[the remark this answers is not in the document]
Comment (nalepae) [reply validator/client/beacon-api/status.go:122]: With the current implementation, we ensure the 
```c.stateValidatorsProvider.GetStateValidators(nil, nil, []string{"active"})``` call
is done (at most once) if at least one validator is in `PENDING`, `PARTIALLY_DEPOSITED` or `DEPOSITED` state.

Unless I'm mistaken, we don't need to compute `validatorStatus.PositionInActivationQueue` (and so we don't need to retrieve `lastActivatedValidatorIndex`) for validators which are not in `PENDING`, `PARTIALLY_DEPOSITED` or `DEPOSITED` state.

If we retrieve `lastActivatedValidatorIndex` before the `for` loop, we may do an extra call to the beacon node (call which retrieve the whole list of validators) for nothing, in the case no input validators is in this set of states.

I'm OK to move the call outside of the loop, I just want to be sure it's OK to do the (maybe useless) extra call to the beacon node.

[the remark this answers is not in the document]
Comment (nalepae) [reply validator/client/beacon-api/status.go:122]: > (doing it inside the loop redundantly makes the same call in each iteration).

It's the whole purpose of `isLastActivatedValidatorIndexRetrieved` variable (not to make the same call in each iteration). We do the call at most once.

[the remark this answers is not in the document]
Comment (nalepae) [reply validator/client/beacon-api/status.go:122]: **Current implementation:**
The `c.stateValidatorsProvider.GetStateValidators(nil, nil, []string{"active"}` BN call is done at most once.
- It is done exactly once if at least one validator is in `PENDING`, `PARTIALLY_DEPOSITED` or `DEPOSITED` state.
- It is NOT done at all if none of the validators is in `PENDING`, `PARTIALLY_DEPOSITED` or `DEPOSITED` state.

**If we retrieve `lastActivatedValidatorIndex` before the for loop:**
The `c.stateValidatorsProvider.GetStateValidators(nil, nil, []string{"active"}` BN call will always be done exactly once, even if not needed (aka none of the validators is in `PENDING`, `PARTIALLY_DEPOSITED` or `DEPOSITED` state.)

> (doing it inside the loop redundantly makes the same call in each iteration).

It's the whole purpose of `isLastActivatedValidatorIndexRetrieved` variable (not to make the same call in each iteration). We do the call at most once.

[the remark this answers is not in the document]
Comment (rkapka) [reply validator/client/beacon-api/status.go:122]: > It's the whole purpose of isLastActivatedValidatorIndexRetrieved variable (not to make the same call in each iteration). We do the call at most once.

I see it now, it's declared before the `for` loop.

**Review by rkapka — APPROVED** [review]

**Comment by nalepae** [comment]
I cannot reproduce the failed `sync_completed` e2e test locally:

```
    --- PASS: TestEndToEnd_MinimalConfig/sync_completed (0.50s)
```

**Comment by nalepae** [comment]
It seems some tests e2e tests fail because of `No space left on device` error: https://buildkite.com/prysmatic-labs/prysm/builds/69708#018577a1-b5ea-40ba-8922-2a0854c91921
(exemple: line 218)

**Pull request merged**